controller/search.go

```diff
@@ -243,3 +247,46 @@ func (c *SearchController) Users(ctx *app.UsersSearchContext) error {
 
 	return ctx.OK(&response)
 }
+
+// Iterate over the WI list and read parent IDs
+// Fetch and load Parent WI in the included list
+func (c *SearchController) enrichWorkItemList(ctx *app.ShowSearchContext, res *app.SearchWorkItemList) {
+	// Need a map of string to UUID to prevent duplicates in place.
+	visitedIDs := map[string]uuid.UUID{}
+	fetchInBatch := []uuid.UUID{}
+	for _, wi := range res.Data {
+		if wi.Relationships != nil && wi.Relationships.Parent != nil && wi.Relationships.Parent.Data != nil {
+			parentIDStr := *wi.Relationships.Parent.Data.ID
+			if _, exist := visitedIDs[parentIDStr]; !exist {
+				id, err := uuid.FromString(parentIDStr)
+				if err != nil {
+					log.Error(ctx, map[string]interface{}{
+						"wi_id": parentIDStr,
+						"err":   err,
+					}, "parent ID is invalid UUID: %s", parentIDStr)
+				}
+				visitedIDs[parentIDStr] = id            // this helps to keep elemtns distinct
+				fetchInBatch = append(fetchInBatch, id) // this helps to fetch WI in batch
+			}
+		}
+	}
+	wis := []*workitem.WorkItem{}
+	err := application.Transactional(c.db, func(appl application.Application) error {
+		var err error
+		wis, err = appl.WorkItems().LoadBatchByID(ctx, fetchInBatch)
```

Comment:
Why not load work items individually?

Reply:
Because `IN` query will give us all items in a single query...